### PR TITLE
DOC: Add clarification for np.array being 0 dimensional

### LINKED ISSUE #19504: DOC: Clarify that `np.array` can be 0-dimensional
## Documentation

From the docstring of `np.array`, its only positional argument is:

> An array, any object exposing the array interface, an object whose __array__ method returns an array, or any (nested) sequence.

From such a description, it is surprising that one can do things like `np.array(1)` (I guess this is done to support 0D arrays).

### PR DESCRIPTION
**Pull request by yashasvimisra2798**
Fixes #19504
I have added the clarification along with an example.
@melissawm please check.

### CODE REVIEW

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -795,7 +795,10 @@
     ----------
     object : array_like
         An array, any object exposing the array interface, an object whose
-        __array__ method returns an array, or any (nested) sequence.
+        __array__ method returns an array, or any (nested) sequence. However
+        if a number is passed as a parameter then it will return the number itself.
```
Comment (Mukulikaa) [inline numpy/core/_add_newdocs.py:799]: Hi @yashasvimisra2798, the lint test is failing because this line has more than 79 characters.

Comment (rossbar) [inline numpy/core/_add_newdocs.py:799]: Also the wording here is a bit confusing - this makes it sound like a pass-through function but in fact a 0-D array is returned (as you note in the next sentence, though the example is incorrect).

My feeling is that the clearest way to add something here would be to note that scalars are array-like and say something like:

> If `object` is a scalar, a 0-dimensional array containing `object` is returned.

**Comment by yashasvimisra2798** [comment]
Ok @Mukulikaa and @rossbar I will make the changes

File: numpy/core/_add_newdocs.py (hunk 1 above)
Comment (yashasvimisra2798) [reply numpy/core/_add_newdocs.py:799]: Hey, @rossbar about the example that I have included here of np.array(2) returning 2, I tried it and I was getting this result so included it here. May you please elaborate more on how this example is wrong?

Comment (rossbar) [reply numpy/core/_add_newdocs.py:799]: > May you please elaborate more on how this example is wrong?

Sure - in the example you are creating a 0-D array with the value 2 which is distinct from the value that you passed into `np.array`:

```python
>>> v = 2
>>> a = np.array(v)
>>> type(v)
int
>>> type(a)
numpy.ndarray
```

Comment (seberg) [reply numpy/core/_add_newdocs.py:799]: A bit of a fly-by comment/thought.  But maybe flipping the logic would be better?  I.e. say: A Python scalar (number), an object convertible to a NumPy array, or any (nested) sequence of these.  <We can continue here explaining what can be converted to NumPy arrays>

Comment (yashasvimisra2798) [reply numpy/core/_add_newdocs.py:799]: Thanks, @rossbar & @seberg for your inputs.

**Comment by yashasvimisra2798** [comment]
Actually, since I was not able to solve conflicts in this PR I have opened another one for this issue. Therefore I'll be closing this PR.

**Pull request closed without merging**